addons/sale/static/src/js/sale_product_field.js

```diff
@@ -147,6 +147,29 @@ export class SaleOrderLineProductField extends ProductLabelSectionAndNoteField {
         return className;
     }
 
+    get label() {
+        let label = super.label;
+        if (label.startsWith(this.translatedProductName)) {
+            // Remove the translated name as it is already shown to the salesman on the SOL.
+            label = label.split("\n", 2).at(1) || "";
+        }
+        return label;
+    }
+
+    get translatedProductName() {
+        return this.props.record.data.translated_product_name;
+    }
+
+    updateLabel(value) {
+        this.props.record.update({
+            name: (
+                this.translatedProductName && value && this.translatedProductName.concat("\n", value)
+                || !value && this.translatedProductName
+                || value
+            ),
+        });
+    }
+
```

Comment:
You should handle the case where `translatedProductName` is empty because the view wasn't updated :eyes:

Reply:
Unresolving this one because if the view isn't updated, `this.translatedProductName` is undefined and if the current user language is the same as the customer language, the product name won't be in the description anymore, which will break existing customer flows.

![image](https://github.com/user-attachments/assets/1027a14e-ada4-4e26-b772-ad256e3bf403)